contracts/cw20-atomic-swap/src/contract.rs

```diff
@@ -362,7 +365,7 @@ mod tests {
         let res = handle(&mut deps, env, HandleMsg::Create(create.clone()));
         match res {
             Ok(_) => panic!("expected error"),
-            Err(StdError::GenericErr { msg, .. }) => {
+            Err(ContractError::EmptyBalance(msg)) => {
```

Comment:
For me the great thing about `ContractError` is not needing to check the message.
We write it declarative in one file and can view/update if we want to.

For tests, we can just assert the proper type (and maybe metadata like length), but no need to compare strings intended for human consumption in a test - just that the expected branch got triggered (eg. failed due to empty balance, not due to unauthorized)